spark/dl/src/main/scala/com/intel/analytics/bigdl/dataset/DataSet.scala

```diff
@@ -394,7 +394,7 @@ object DataSet {
     if (isInOrder) {
       require(classTag[T] == classTag[Sample[_]],
         "DataSet.sortData: Only support sort for sample input")
-      data.sortBy(a => a.asInstanceOf[Sample[_]].feature().nElement())
+      data.sortBy(a => a.asInstanceOf[Sample[_]].nElement()._1)
```

Comment:
@zhangxiaoli73 We only need the length (i.e., the size of the first dimension) of the feature here, not the number of elements